### PR TITLE
[4][SQL] - batch copy

### PR DESCRIPTION
**Pull request by alikon**
### Summary of Changes
cause of mysql vs postgresql update join different syntax:

- issued different SQL query  depending on the db server
- converted SQL to prepared statements

### Testing Instructions
batch copy an article (on postgresql)

### Actual result BEFORE applying this Pull Request
on postgresql  you got an error

### Expected result AFTER applying this Pull Request
with both supported db servers you are able to batch copy an article 

### Additional comment
this issue is present from 3.9.11, and it has been introduced with #24736 by myself :sob: 
i need more time to prepare a patch for 3 cause of mssql and i don't have a mssql environment to test

### CODE REVIEW

File: libraries/src/MVC/Model/AdminModel.php [hunk 1]
```diff
@@ -510,15 +510,31 @@ protected function batchCopy($value, $pks, $contexts)
...
+
+				$query->join(
+					'INNER',
+					$db->quoteName('#__assets', 's'),
+					$db->quoteName('s.id') . ' = :oldassetid'
+				)
+					->where($db->quoteName('t.id') . ' = :assetid')
+					->bind(':oldassetid', $oldAssetId, ParameterType::INTEGER)
+					->bind(':assetid', $this->table->asset_id, ParameterType::INTEGER);
+
```
Comment (richard67) [inline libraries/src/MVC/Model/AdminModel.php:537]:
```suggestion
```
Remove one of the 2 empty lines in a row to fix Drone PHPCS.

**Comment by VladimirAus** [comment]
I have tested this item :white_check_mark: successfully on dc488b42f109483ec2c72f5960c946663bc91543<br /><br />* Tested functionality on mysql: 5.6: `pass`
* Tested functionality on postgres: 12.4: `failed`
* Tested functionality on postgres: 12.4 with patch: `pass`<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31036">issues.joomla.org/tracker/joomla-cms/31036</a>.</sub>

**Comment by waader** [comment]
I have tested this item :white_check_mark: successfully on dc488b42f109483ec2c72f5960c946663bc91543<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31036">issues.joomla.org/tracker/joomla-cms/31036</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31036">issues.joomla.org/tracker/joomla-cms/31036</a>.</sub>

**Comment by HLeithner** [comment]
There is no way around this sql driver if statement?

**Comment by alikon** [comment]
not that i'm aware of...
the same happens on com_finder where we are using more complex sql, with a separate driver and even with if  like here
https://github.com/joomla/joomla-cms/blob/4.0-dev/administrator/components/com_finder/src/Indexer/Indexer.php#L534
https://github.com/joomla/joomla-cms/blob/4.0-dev/administrator/components/com_finder/src/Indexer/Indexer.php#L558
https://github.com/joomla/joomla-cms/blob/4.0-dev/administrator/components/com_finder/src/Indexer/Indexer.php#L730

**Pull request merged**